lib/ecto/type.ex

```diff
@@ -634,6 +646,7 @@ defmodule Ecto.Type do
   defp of_base_type?(:binary, term),     do: is_binary(term)
   defp of_base_type?(:string, term),     do: is_binary(term)
   defp of_base_type?(:map, term),        do: is_map(term) and not Map.has_key?(term, :__struct__)
+  defp of_base_type?({:map, _}, _),      do: false # Always handled explicitly.
```

Comment:
I wasn't sure what to put here, so I just copied what we had for `{:array, type}`